Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -29,23 +29,29 @@ actor ScopeGenerator {
         propertiesToGenerate: [ScopeGenerator],
         receivedProperties: Set<Property>
     ) {
-        self.instantiable = instantiable
+        if let property {
+            scopeData = .property(
```

Comment:
I've now created a `ScopeData` enum to encapsulate the different kinds of scopes that exist that we need to generate for. Scroll down to find the definition.